tests/bofire/strategies/test_sobo.py

```diff
@@ -178,3 +180,68 @@ def test_get_acqf_input(acqf, num_experiments, num_candidates):
         num_candidates,
         len(set(chain(*names.values()))),
     )
+
+
+def test_custom_get_objective():
+    def f(samples, callables, weights, X):
+        outputs_list = []
+        for c, w in zip(callables, weights):
+            outputs_list.append(c(samples, None) ** w)
+        samples = torch.stack(outputs_list, dim=-1)
+
+        return (samples[..., 0] + samples[..., 1]) * (samples[..., 0] * samples[..., 1])
+
+    benchmark = DTLZ2(3)
+    data_model = data_models.CustomSoboStrategy(
+        domain=benchmark.domain, acquisition_function=qNEI()
+    )
+    strategy = CustomSoboStrategy(data_model=data_model)
+    strategy.f = f
+    generic_objective = strategy._get_objective()
+    assert isinstance(generic_objective, GenericMCObjective)
+
+
+def test_custom_get_objective_invalid():
+    benchmark = DTLZ2(3)
+    data_model = data_models.CustomSoboStrategy(
+        domain=benchmark.domain, acquisition_function=qNEI()
+    )
+    strategy = CustomSoboStrategy(data_model=data_model)
+
+    with pytest.raises(ValueError):
+        strategy._get_objective()
+
+
+def test_custom_dumps_loads():
+    def f(samples, callables, weights, X):
+        outputs_list = []
+        for c, w in zip(callables, weights):
+            outputs_list.append(c(samples, None) ** w)
+        samples = torch.stack(outputs_list, dim=-1)
+
+        return (samples[..., 0] + samples[..., 1]) * (samples[..., 0] * samples[..., 1])
+
+    benchmark = DTLZ2(3)
+    data_model1 = data_models.CustomSoboStrategy(
+        domain=benchmark.domain, acquisition_function=qNEI()
+    )
+    strategy1 = CustomSoboStrategy(data_model=data_model1)
+    strategy1.f = f
+    f_str = strategy1.dumps()
+
+    data_model2 = data_models.CustomSoboStrategy(
+        domain=benchmark.domain, acquisition_function=qNEI()
+    )
+    strategy2 = CustomSoboStrategy(data_model=data_model2)
+    strategy2.loads(f_str)
+    assert isinstance(strategy2.f, type(f))
```

Comment:
do we want to test here that evaluating objective of `strategy` amounts to the same as evaluating the loaded objective of `strategy2`?